internal/scraper/scraper_test.go

```diff
@@ -492,6 +493,284 @@ func setupTCPServerWithSSL(t *testing.T) (string, func()) {
 	}
 }
 
+func TestMakeTimeseries(t *testing.T) {
+	testTime := time.Now()
+	testValue := 42.0
+	testcases := map[string]struct {
+		t      time.Time
+		value  float64
+		labels []prompb.Label
+	}{
+		"no labels": {
+			t:      testTime,
+			value:  testValue,
+			labels: []prompb.Label{},
+		},
+		"one label": {
+			t:     testTime,
+			value: testValue,
+			labels: []prompb.Label{
+				{Name: "name", Value: "value"},
+			},
+		},
+		"many labels": {
+			t:     testTime,
+			value: testValue,
+			labels: []prompb.Label{
+				{Name: "name 1", Value: "value 1"},
+				{Name: "name 2", Value: "value 2"},
+				{Name: "name 3", Value: "value 3"},
+				{Name: "name 4", Value: "value 4"},
+			},
+		},
+	}
+
+	for name, tc := range testcases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			actual := makeTimeseries(tc.t, tc.value, tc.labels...)
+			require.Equal(t, len(tc.labels), len(actual.Labels))
+			require.Equal(t, tc.labels, actual.Labels)
+			require.Equal(t, 1, len(actual.Samples))
+			require.Equal(t, tc.t.UnixNano()/1e6, actual.Samples[0].Timestamp)
+			require.Equal(t, tc.value, actual.Samples[0].Value)
+		})
+	}
+}
+
+func TestAppendDtoToTimeseries(t *testing.T) {
```

Comment:
@mem I think there's a typo here. Feels like it should read `TestAppendToTimeseries`

Reply:
oh ok, my bad - thanks for your reply. So can I ask - what is "Dto"? What does it mean? Thanks!

Reply:
Data Transfer Object. In this case it's the Prometheus representation of the metric that is transmitted between processes.